### DISCUSSION TITLE
How to search through the whole path, as if using Everything

### DISCUSSION THREAD

**Discussion by TheBestPessimist** [Flow-Launcher/Flow.Launcher#1800] [Q&A] [answered]
I have a folder `D:/all/wallpapers/artstation/pics and nested folders here`

In Everything, if i enable `Search` -> `Match Path` (<kbd>Ctrl</kbd>+<kbd>U</kbd>) i can simply type `wall arts` and it shows me all the correct files.
With FL, i have to type `wall*/arts` to have the same:

![image](https://user-images.githubusercontent.com/4482210/212491977-9ef92e84-78ce-4d03-abba-df11afadd0cf.png)

If i type `wall arts` i get the basic google search instead:
![image](https://user-images.githubusercontent.com/4482210/212492091-94ed915c-affe-44eb-a2c2-da983a2ac0f6.png)

Is there  a way to make FL work like Everything? Or pass the literal search term that i typed to Everything?

**Comment by TheBestPessimist** [comment]
Here's another example of how Everything query looks, compared to FL. In FL it is extremely cumbersome to search in a "full path":
![image](https://user-images.githubusercontent.com/4482210/212492202-5cd91774-2e61-4c94-b8d8-3178ffd09383.png)

**Comment by taooceros** [comment]
Well I don't really know how everything is handling this.....Though, what about marking all ` ` as `*` for everything query? @Flow-Launcher/team

**Reply by TheBestPessimist** [reply]
If it helps with brainstorming, this might be related: https://github.com/Flow-Launcher/Flow.Launcher/issues/1805

**Reply by TheBestPessimist** [reply]
basically, in everything i also use the `Match Path` flag:
![image](https://user-images.githubusercontent.com/4482210/213981517-d5f1260b-9e78-4173-97d1-ea8fbd4fefa8.png)

which make (to my eyes) the following changes:
- search works in the full path, not just in the file name
- space (` `) works like regexp `.*` when matching through the full path

**Reply by TheBestPessimist** [reply]
One more note: from what i can see, `*` in FL does not match folder delimiter (`\`). 
That's why in my first example i cannot use `wall*arts` but i am forced to use `wall*\*arts` instead, So even with your suggestion, i'm not sure FL will do what i want.

I'm thinking that FL should instead should use the `Match Path` flag. Maybe the SDK gives an option for this?

Or even better, could FL simply pass the whole query string to Everything, and let Everything do its job of parsing it? This would probably make using filters or other functionality (eg. size greater than, etc) easier for the FL/Everything  user. Basically the same query used directly in Everything will work from FL

**Comment by TheBestPessimist** [comment]
Looking at the SDK, i think what we want is this: https://www.voidtools.com/support/everything/sdk/everything_setmatchpath/

**Comment by TheBestPessimist** [comment, marked as answer]
Done in https://github.com/Flow-Launcher/Flow.Launcher/pull/1848
